### PR TITLE
Added 5 new CRUD tests for Config Template via API.

### PR DESCRIPTION
**Pull request by omaciel**
I have added 5 new 'CRUD' tests to exercise Configuration Templates via the API.

### CODE REVIEW

**Comment by omaciel** [comment]
```bash
(robottelo)┌[omaciel☮Ogs-MacBook-Pro.local]-(~/hacking/robottelo)-[git://master ✗]-
└> nosetests -v tests/foreman/api/test_config_template.py
@Test: Call the "build_pxe_default" path. ... ok
@Test: Associate a config template with organizations. ... ok
@Test: Create configuration template providing an invalid name. ... ok
@Test: Create configuration template then update its name to an ... ok
@Test: Create a configuration template providing the initial name. ... ok
@Test: Create configuration template and then delete it. ... ok
@Test: Create configuration template providing the initial name, ... ok

----------------------------------------------------------------------
Ran 7 tests in 19.915s

OK
```

File: tests/foreman/api/test_config_template.py [hunk 1]
```diff
@@ -63,3 +63,78 @@ def test_config_template_orgs(self):
...
+        @Feature: Configuration Template
+        """
+        for name in valid_data_list():
+            with self.subTest(name):
+                c_temp = entities.ConfigTemplate(name=name).create()
+                self.assertEqual(name, c_temp.name)
+
+    def test_negative_create_name(self):
+        """@Test: Create configuration template providing an invalid name.
+        set.
```
Comment (sthirugn) [inline tests/foreman/api/test_config_template.py]: Remove set.

**Comment by sthirugn** [comment]
Ack pending minor comment

**Comment by JacobCallahan** [comment]
ACK

**Comment by omaciel** [comment]
@sthirugn sorry about that. Done

File: tests/foreman/api/test_config_template.py [hunk 2]
```diff
@@ -63,3 +63,77 @@ def test_config_template_orgs(self):
...
+                c_temp = entities.ConfigTemplate(id=c_temp.id).read()
+                self.assertNotEqual(c_temp.name, new_name)
+
+    def test_positive_delete(self):
+        """@Test: Create configuration template and then delete it.
+
+        @Assert: Configuration Template is successfully deleted.
+
+        @Feature: Configuration Template
+
```
Comment (elyezer) [inline tests/foreman/api/test_config_template.py]: This blank lines could be avoided :smile:

**Comment by elyezer** [comment]
:+1:

**Pull request merged**